CMakeLists.txt

```diff
@@ -1,264 +1,84 @@
-set (SC_MACHINE_THIRDPARTY_PATH "${SC_MACHINE_ROOT}/thirdparty")
-
-option (SC_AUTO_TEST "Flag to build for automation testing" OFF)
-option (SC_BUILD_SCTP "Flag to turn on/off sctp protocol support" ON)
-option (SC_BUILD_TESTS "Flag to build unit tests" OFF)
-option (SC_BUILD_BENCH "Flag to build benchmark" OFF)
-option (SC_COVERAGE "Flag to generate coverage report" OFF)
-option (SC_CLANG_FORMAT_CODE "flag to add clangformat and clangformat_check targets" OFF)
-
-# Coverage
-if(${SC_COVERAGE} AND CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -O0 --coverage")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -O0 --coverage")
-endif()
-
-if(${SC_CLANG_FORMAT_CODE})
-	include(${CMAKE_CURRENT_LIST_DIR}/cmake/ClangFormat.cmake)
-endif()
-
-
-set(SC_USE_SANITIZER "none" CACHE STRING "Build with specified sanitizer")
-set_property(CACHE SC_USE_SANITIZER PROPERTY STRINGS none address memory)
-
-
-if (${SC_BUILD_ARM64})
-    check_c_compiler_flag("-arch arm64" IS_ARM64_SUPPORTED)
-    if (${IS_ARM64_SUPPORTED})
-        set (CMAKE_OSX_ARCHITECTURES "arm64" CACHE STRING "Used architecture" FORCE)
-        message ("-- Architecture: arm64")
-    else()
-        message (FATAL_ERROR "Arm64 platform is not supported by compiler")
-    endif()
-endif()
-
-# codegen
-if (${UNIX})
-    set(SC_CODEGEN_TOOL "${SC_BIN_PATH}/sc-code-generator")
-elseif (${WIN32})
-    set(SC_CODEGEN_TOOL "${SC_BIN_PATH}/sc-code-generator.exe")
-endif()
-include("${SC_MACHINE_ROOT}/cmake/codegen.cmake")
-
-find_package(Java)
+set(SC_MACHINE_CONFIG "${SC_MACHINE_ROOT}/config")
+configure_file("${SC_MACHINE_CONFIG}/sc-machine.ini.in" "${SC_MACHINE_CONFIG}/sc-machine.ini")
+logger(SC_MACHINE_LOG "Memory messages logger" "${SC_MACHINE_ROOT}/log/sc-machine.log" Debug Console)
 
 set(CMAKE_MODULE_PATH "${SC_MACHINE_ROOT}/cmake")
-
+set(SC_BIN_PATH "${SC_MACHINE_ROOT}/bin")
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${SC_BIN_PATH})
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${SC_BIN_PATH})
+set(SC_EXTENSIONS_DIRECTORY "${SC_BIN_PATH}/extensions")
+set(SC_MACHINE_THIRDPARTY_PATH "${SC_MACHINE_ROOT}/thirdparty")
 
-set(SC_EXTENSION_DIRECTORY "${SC_BIN_PATH}/extensions")
+option(SC_AUTO_TEST "Flag to build for automation testing" OFF)
+option(SC_BUILD_TESTS "Flag to build unit tests" ON)
+option(SC_BUILD_BENCH "Flag to build benchmark" OFF)
+option(SC_BUILD_SCTP "Flag to turn on/off sctp protocol support" ON)
 
-# for multi-config builds (e.g. msvc)
-foreach( OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES} )
-    string( TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG )
-    set( CMAKE_RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${SC_BIN_PATH} )
-    set( CMAKE_LIBRARY_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${SC_BIN_PATH} )
-endforeach( OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES )
+default_coverage(SC_COVERAGE "Flag to generate coverage report" OFF)
+option(SC_CLANG_FORMAT_CODE "Flag to add clangformat and clangformat_check targets" OFF)
+no_sanitizer(SC_USE_SANITIZER "Build with specified sanitizer")
```

Comment:
I don't think you can read this easily. It would make more sense to pass the mode name to the `sanitizer` macros rather than creating several macros for different sanitizer types.